enclave/core/sgx/calls.c

```diff
@@ -585,9 +593,31 @@ oe_result_t oe_call_host_function_by_table_id(
     args->result = OE_UNEXPECTED;
 
     /* Call the host function with this address */
-    // TODO: for switchessless calls, push the job (wrapped in args) to an
-    // available worker thread, and wait for result
-    // if (!switchless)
+    if (switchless && oe_is_switchless_initialized())
+    {
+        oe_result_t post_result = oe_post_switchless_ocall(args);
+
+        // Fall back to regular OCALL if host worker threads are unavailable
+        if (post_result == OE_CONTEXT_SWITCHLESS_OCALL_MISSED)
+            OE_CHECK(
+                oe_ocall(OE_OCALL_CALL_HOST_FUNCTION, (uint64_t)args, NULL));
+        else
+        {
+            OE_CHECK(post_result);
+            // Wait until args.result is set by the host worker.
+            while (true)
+            {
+                OE_ATOMIC_MEMORY_BARRIER_ACQUIRE();
+                if (__atomic_load_n(&args->result, __ATOMIC_SEQ_CST) !=
```

Comment:
Do you think we should use the memorder parameter of __atomic_load_n instead of calling OE_ATOMIC_MEMORY_BARRIER_ACQUIRE explicitly?

Reply:
They are functionally the same - probably translated into nop in both cases. I don't have preference one over the other.